src/components/Search/MainPage.js

```diff
@@ -1,84 +1,79 @@
 import React, {Component} from 'react';
 import { Route, Link } from "react-router-dom";
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
 
+import * as types from '../../actions/actionTypes';
 import Constants from '../Common/constants';
 import Header from '../Header/Header';
 import SearchResults from '../Search/SearchResults';
+import * as mainActions from '../../actions/mainActions';
 
 require('es6-promise').polyfill();
 require('isomorphic-fetch');
 
-export default class MainPage extends Component {
+export class MainPage extends Component {
     constructor(props) {
         super(props);
-        this.state = {
-            movies: [],
-            titleActive: true,
-            sortByReleaseDate: true,
-            searchText: ''
-        };
         this.handleTitleClick = this.handleTitleClick.bind(this);
         this.handleGenreClick = this.handleGenreClick.bind(this);
-        this.handleSearchClick = this.handleSearchClick.bind(this);
+        this.refreshSearchResults = this.refreshSearchResults.bind(this);
         this.handleReleaseDateClick = this.handleReleaseDateClick.bind(this);
         this.handleRatingClick = this.handleRatingClick.bind(this);
         this.valueChangeHandler = this.valueChangeHandler.bind(this);
     }
     refreshSearchResults() {
-        let titleOrGenre = (this.state.titleActive === true) ? 'title' : 'genres';
-        let dateOrVote = (this.state.sortByReleaseDate === true) ? 'release_date' : 'vote_average';
-        let urlStr = `${Constants.baseURL}?search=${this.state.searchText}&searchBy=${titleOrGenre}&sortBy=${dateOrVote}&sortOrder=desc`;
-        this.fetchSearchResults(urlStr).then(response => {
-            this.populateState(response.data);
-        });
-    }
-    populateState(response) {
-        this.setState({
-            movies: response
-        });
-    }
-    fetchSearchResults(urlStr) {
-        return fetch(urlStr).then(res => res.json());
-    }
-    handleSearchClick() {
-        this.refreshSearchResults();
+        let titleOrGenre = (this.props.titleActive === true) ? 'title' : 'genres';

```

Comment:
These values are not reassigned so it's better to use `const`

Reply:
Okay